src/fp6.rs

```diff
@@ -266,76 +310,239 @@ impl Fp6 {
         // are zero and the c0 coefficient is lexicographically largest,
         // then this element is lexicographically largest.
 
-        self.c2.lexicographically_largest()
+        self.c5.lexicographically_largest()
+            | (self.c5.is_zero() & self.c4.lexicographically_largest())
+            | (self.c4.is_zero() & self.c3.lexicographically_largest())
+            | (self.c3.is_zero() & self.c2.lexicographically_largest())
             | (self.c2.is_zero() & self.c1.lexicographically_largest())
             | (self.c1.is_zero() & self.c0.lexicographically_largest())
     }
 
     #[inline]
-    /// Computes the multiplication of two Fp6 elements
-    pub const fn mul(&self, other: &Fp6) -> Fp6 {
-        let aa = (&self.c0).mul(&other.c0);
-        let bb = (&self.c1).mul(&other.c1);
-        let cc = (&self.c2).mul(&other.c2);
+    /// Computes the multiplication of an Fp6 element with an Fp element
+    pub const fn mul_by_fp(&self, other: &Fp) -> Fp6 {
+        // All helper values computed below are seen as u128 after modular reduction.
+        // This allows for a faster computation of the result coordinates,
+        // by computing all operations in /ZZ, and then finally converting the
+        // values back to Fp through a modular reduction.
+        //
+        // The reduction uses `reduce_u96()` as all final values are less than 96 bits.
```

Comment:
Does this apply for this function? There aren't really any helper values below, and internally, `Fp::mul` uses `reduce_u128`, right?